File: AwesomeStringConverter.java
import java.util.*;

enum FromStatus
{
    NULL,
    STR, BIN, DEC, HEX
}

public class AwesomeStringConverter
{
    private String encoding = "UTF-8";
    private int bufferSize = 65536;
    private int bytesSize = 65536;

    private FromStatus from = FromStatus.NULL;

    private String source = "";

    String getEncoding()
    {
        return encoding;
    }
    void setUtf8()
    {
        setEncoding("UTF-8");
    }
    void setUtf16()
    {
        setEncoding("UTF-16");
    }
    void setUtf32()
    {
        setEncoding("UTF-32");
    }
    void setEncoding(String en)
    {
        encoding = en;
    }


    //인코딩된 문자열로 반환
    String toStr()
    {
        try
        {
            switch(from)
            {
                case STR:
                    return source;

                case BIN:
                {
                    StringTokenizer tokenizer = new StringTokenizer(source);
                    byte[] bytes = new byte[tokenizer.countTokens()];
                    int i = 0;
                    while(tokenizer.hasMoreTokens())
                    {
                        bytes[i] = (byte)Integer.parseInt(tokenizer.nextToken(), 2);
                        i++;
                    }
                    return new String(bytes, encoding);
                }

                case DEC:
                {
                    StringTokenizer tokenizer = new StringTokenizer(source);
                    byte[] bytes = new byte[tokenizer.countTokens()];
                    int i = 0;
                    while(tokenizer.hasMoreTokens())
                    {
                        bytes[i] = (byte)Integer.parseInt(tokenizer.nextToken(), 10);
                        i++;
                    }
                    return new String(bytes, encoding);
                }

                case HEX:
                {
                    StringTokenizer tokenizer = new StringTokenizer(source);
                    byte[] bytes = new byte[tokenizer.countTokens()];
                    int i = 0;
                    while(tokenizer.hasMoreTokens())
                    {
                        bytes[i] = Byte.parseByte(tokenizer.nextToken(), 16);
                        i++;
                    }
                    return new String(bytes, encoding);
                }
            }

        }
        catch(Exception e)
        {return e.toString();}
        return "error";
    }


    //2진수 문자열로 반환
    String toBin()
    {
        try
        {
            byte[] bytes = source.getBytes(encoding);

            switch(from)
            {
                case STR:
                {
                    StringBuilder buffer = new StringBuilder(bufferSize);

                    for(byte e : bytes)
                        buffer.append(
                                String.format("%8s",Integer.toBinaryString(e&0xff))
                                        .replace(' ', '0')
                        ).append(' ');

                    return buffer.toString();
                }

                case BIN:
                    return source;

                case DEC:
                {
                    StringTokenizer tokenizer = new StringTokenizer(source);
                    StringBuilder buffer = new StringBuilder(bufferSize);

                    while(tokenizer.hasMoreTokens())
                    {
                        int e = Integer.parseInt(tokenizer.nextToken());
                        buffer.append(
                                String.format("%8s", Integer.toBinaryString(e))
                                        .replace(' ', '0')
                        ).append(' ');
                    }
                    return buffer.toString();
                }


                case HEX:
                {
                    StringTokenizer tokenizer = new StringTokenizer(source);
                    StringBuilder buffer = new StringBuilder(bufferSize);

                    while(tokenizer.hasMoreTokens())
                    {;
                        int e = Integer.parseInt("EC" +tokenizer.nextToken(), 16);

                        buffer.append(
                                String.format("%8s", Integer.toBinaryString(e&0xff))
                                        .replace(' ', '0')
                        ).append(' ');
                    }
                    return buffer.toString();
                }
            }
        }
        catch(Exception e)
        {return e.toString();}
        return "";
    }


    //10진수 문자열로 반환
    String toDec()
    {
        try
        {
            switch(from)
            {
                case STR:
                {
                    StringBuilder buffer = new StringBuilder(bufferSize);

                    byte[] bytes = source.getBytes(encoding);
                    for(byte e : bytes)
                        buffer.append(e&0xff).append(' ');

                    return buffer.toString();
                }


                case BIN:
                {
                    StringBuilder buffer = new StringBuilder(bufferSize);
                    StringTokenizer tokenizer = new StringTokenizer(source);

                    while(tokenizer.hasMoreTokens())
                    {
                        int e = Integer.parseInt(tokenizer.nextToken(), 2);
                        buffer.append(String.format("%d",e))
                                .append(' ');
                    }
                    return buffer.toString();
                }


                case DEC:
                    return source;

                case HEX:
                {
                    StringBuilder buffer = new StringBuilder(bufferSize);
                    StringTokenizer tokenizer = new StringTokenizer(source);

                    while(tokenizer.hasMoreTokens())
                    {
                        int e = Integer.parseInt(tokenizer.nextToken(), 16);
                        buffer.append(String.format("%d",e))
                                .append(' ');
                    }
                    return buffer.toString();
                }
            }
        }
        catch(Exception e)
        {return e.toString();}
        return "";
    }


    //16진수 문자열로 변경
    String toHex()
    {
        try
        {
            switch(from)
            {
                case STR:
                {
                    StringBuilder buffer = new StringBuilder(bufferSize);
                    byte[] bytes = source.getBytes(encoding);

                    for(byte e : bytes)
                        buffer.append(String.format("%02X", e))
                                .append(' ');

                    return buffer.toString();
                }

                case BIN:
                {
                    StringBuilder buffer = new StringBuilder(bufferSize);
                    StringTokenizer tokenizer = new StringTokenizer(source);

                    while(tokenizer.hasMoreTokens())
                    {
                        int e = Integer.parseInt(tokenizer.nextToken(), 2);
                        buffer.append(String.format("%02X",e))
                                .append(' ');
                    }
                    return buffer.toString();
                }

                case DEC:
                {
                    StringBuilder buffer = new StringBuilder(bufferSize);
                    StringTokenizer tokenizer = new StringTokenizer(source);

                    while(tokenizer.hasMoreTokens())
                    {
                        int e = Integer.parseInt(tokenizer.nextToken(), 10);
                        buffer.append(String.format("%02X",e))
                                .append(' ');
                    }
                    return buffer.toString();
                }

                case HEX:
                    return source;
            }
        }
        catch(Exception e)
        {return e.toString();}
        return "";
    }

    public AwesomeStringConverter fromString(String s)
    {
        from = FromStatus.STR;
        source = s;

        return this;
    }
    public AwesomeStringConverter fromBin(String s)
    {
        from = FromStatus.BIN;
        source = s;

        return this;
    }
    public AwesomeStringConverter fromDec(String s)
    {
        from = FromStatus.DEC;
        source = s;

        return this;
    }
    public AwesomeStringConverter fromHex(String s)
    {
        from = FromStatus.HEX;
        source = s;

        return this;
    }
}

File: AwesomeStringConverter.kt
package com.myyrakle.awesomeconverter

import java.nio.charset.Charset
import java.util.*
import kotlin.text.StringBuilder

enum class FromStatus
{
    NULL,
    STR, BIN, DEC, HEX
}

open class AwesomeStringConverter()
{
    var encoding: Charset = charset("UTF-8")
    var debuggingMode: Boolean = false

    private var from = FromStatus.NULL
    private var source = ""

    private var bufferSize = 65536
    private var byteSize = 65536

    //인코딩을 설정합니다.
    fun setUtf8()
    {
        encoding = charset("UTF-8")
    }
    fun setUtf16()
    {
        encoding = charset("UTF-16")
    }
    fun setUtf32()
    {
        encoding = charset("UTF-32")
    }
    fun setEucKr()
    {
        encoding = charset("euc-kr")
    }
    fun setEncoding(encodingName:String )
    {
        encoding = charset(encodingName)
    }

    //변환될 값과 타입을 받습니다.
    fun fromString(s: String): AwesomeStringConverter
    {
        from = FromStatus.STR
        source = s

        return this
    }
    fun fromBin(s: String): AwesomeStringConverter
    {
        from = FromStatus.BIN
        source = s

        return this
    }
    fun fromDec(s: String): AwesomeStringConverter
    {
        from = FromStatus.DEC
        source = s

        return this
    }
    fun fromHex(s: String): AwesomeStringConverter
    {
        from = FromStatus.HEX
        source = s

        return this
    }

    @kotlin.ExperimentalUnsignedTypes
    fun toStr(): String=
        when(from)
        {
            FromStatus.STR ->source

            FromStatus.BIN, FromStatus.DEC, FromStatus.HEX ->
            {
                val radix = when(from)
                {
                    FromStatus.BIN -> 2
                    FromStatus.DEC -> 10
                    FromStatus.HEX -> 16
                    else -> 0
                }

                var tokenizer = StringTokenizer(source)
                var bytes = Array<Byte>(tokenizer.countTokens(), {0})
                var i = 0
                while(tokenizer.hasMoreTokens())
                {
                    bytes[i] = try {tokenizer.nextToken().toUByte(radix).toByte()} catch(e:Exception) {0.toByte()}
                    i++
                }

                bytes.toByteArray().toString(encoding)
            }

            FromStatus.NULL -> "null"
        }


    @kotlin.ExperimentalUnsignedTypes
    fun toBin(): String=
        when(from)
        {
            FromStatus.BIN -> source

            FromStatus.STR ->
            {
                var builder = StringBuilder(bufferSize)
                val bytes = source.toByteArray(encoding)

                for(e in bytes)
                {
                    val formatted =
                        try
                        {
                            String.format("%8s", Integer.toBinaryString(e.toInt().and(0xff))).
                                replace(' ', '0')
                        }
                        catch(e:Exception){"???"}

                    builder.append(formatted).append(' ')
                }

                builder.toString()
            }

            FromStatus.DEC, FromStatus.HEX ->
            {
                val radix = when(from)
                {
                    FromStatus.DEC -> 10
                    FromStatus.HEX -> 16
                    else -> 0
                }

                var tokenizer = StringTokenizer(source)
                var builder = StringBuilder()
                while (tokenizer.hasMoreTokens())
                {
                    val formatted = try
                    {
                        val byte: Int = tokenizer.nextToken().toInt(radix)

                        if(byte > UByte.MAX_VALUE.toInt())
                            String.format("%s", Integer.toBinaryString(byte))
                        else
                            String.format("%8s", Integer.toBinaryString(byte)).replace(' ', '0')
                    }
                    catch(e:Exception)
                    {"???"}

                    builder.append(formatted).append(' ')
                }
                builder.toString()
            }

            FromStatus.NULL -> "null"
        }

    @kotlin.ExperimentalUnsignedTypes
    fun toDec(): String=
        when(from)
        {
            FromStatus.DEC -> source

            FromStatus.STR ->
            {
                var builder = StringBuilder(bufferSize)
                val bytes = source.toByteArray(encoding)

                for(e in bytes)
                {
                    builder.
                        append(e.toUByte().toString()).
                        append(' ')
                }

                builder.toString()
            }

            FromStatus.BIN, FromStatus.HEX ->
            {
                val radix = when(from)
                {
                    FromStatus.BIN -> 2
                    FromStatus.HEX -> 16
                    else -> 0
                }

                var tokenizer = StringTokenizer(source)
                var builder = StringBuilder(bufferSize)
                while (tokenizer.hasMoreTokens())
                {
                    val formatted = try
                    {
                        tokenizer.nextToken().toLong(radix).toString()
                    }
                    catch(e:Exception)
                    {"???"}

                    builder.append(formatted).append(' ')
                }
                builder.toString()
            }

            FromStatus.NULL -> "null"
        }


    fun toHex(): String=
        when(from)
        {
            FromStatus.HEX -> source

            FromStatus.STR ->
            {
                var builder = StringBuilder(bufferSize)
                val bytes = source.toByteArray(encoding)

                for(e in bytes)
                {
                    val formatted = try { String.format("%X", e) }
                    catch(e:Exception) {"???"}

                    builder.
                        append(formatted).
                        append(' ')
                }

                builder.toString()
            }

            FromStatus.BIN, FromStatus.DEC ->
            {
                val radix = when(from)
                {
                    FromStatus.BIN -> 2
                    FromStatus.DEC -> 10
                    else -> 0
                }

                var tokenizer = StringTokenizer(source)
                var builder = StringBuilder(bufferSize)
                while (tokenizer.hasMoreTokens())
                {
                    val formatted = try{
                        String.format("%02X", tokenizer.nextToken().toLong(radix))
                    }
                    catch(e:Exception) { "???" }
                    builder.append( formatted ).append(' ')
                }
                builder.toString()
            }

            FromStatus.NULL -> "null"
        }
}